In [89]:
import pandas as pd
import os 


In [90]:
prefix = r'archive/'

intakesCsv = os.path.join(prefix, r'aac_intakes.csv')

intakesOutcomesCsv = os.path.join(prefix, r'aac_intakes_outcomes.csv')

outcomesCsv = os.path.join(prefix, r'aac_outcomes.csv')

intakesCsv, intakesOutcomesCsv, outcomesCsv

('archive/aac_intakes.csv',
 'archive/aac_intakes_outcomes.csv',
 'archive/aac_outcomes.csv')

In [170]:
intakes = pd.read_csv(intakesCsv, index_col=0)
intakes.head()


animal_id animal_type                                  breed  \
age_upon_intake                                                                
8 years           A706918         Dog               English Springer Spaniel   
11 months         A724273         Dog                            Basenji Mix   
4 weeks           A665644         Cat                 Domestic Shorthair Mix   
4 years           A682524         Dog  Doberman Pinsch/Australian Cattle Dog   
2 years           A743852         Dog                 Labrador Retriever Mix   

                       color                 datetime  \
age_upon_intake                                         
8 years          White/Liver  2015-07-05T12:59:00.000   
11 months        Sable/White  2016-04-14T18:43:00.000   
4 weeks               Calico  2013-10-21T07:59:00.000   
4 years             Tan/Gray  2014-06-29T10:38:00.000   
2 years            Chocolate  2017-02-18T12:46:00.000   

                               datetime2                      found_location  \
age_upon_intake                                                                
8 years          2015-07-05T12:59:00.000    9409 Bluegrass Dr in Austin (TX)   
11 months        2016-04-14T18:43:00.000  2818 Palomino Trail in Austin (TX)   
4 weeks          2013-10-21T07:59:00.000                         Austin (TX)   
4 years          2014-06-29T10:38:00.000       800 Grove Blvd in Austin (TX)   
2 years          2017-02-18T12:46:00.000                         Austin (TX)   

                intake_condition      intake_type     name sex_upon_intake  
age_upon_intake                                                             
8 years                   Normal            Stray    Belle   Spayed Female  
11 months                 Normal            Stray  Runster     Intact Male  
4 weeks                     Sick            Stray      NaN   Intact Female  
4 years                   Normal            Stray      Rio   Neutered Male  
2 years                   Normal  Owner Surrender     Odin   Neutered Male

In [171]:
intakes.describe()


animal_id animal_type                   breed        color  \
count      80187       80187                   80187        80187   
unique     72365           5                    2166          529   
top      A721033         Dog  Domestic Shorthair Mix  Black/White   
freq          13       45743                   23519         8340   

                       datetime                datetime2 found_location  \
count                     80187                    80187          80187   
unique                    57055                    57055          36772   
top     2016-09-23T12:00:00.000  2016-09-23T12:00:00.000    Austin (TX)   
freq                         64                       64          14443   

       intake_condition intake_type   name sex_upon_intake  
count             80187       80187  55603           80186  
unique                8           5  14731               5  
top              Normal       Stray  Bella     Intact Male  
freq              70520       56280    357           25488

In [211]:
intakes.shape

(80187, 11)

In [93]:
intakesOutcomes = pd.read_csv(intakesOutcomesCsv, index_col=0)
intakesOutcomes.head()

animal_id_outcome        date_of_birth outcome_subtype  \
age_upon_outcome                                                          
10 years                   A006100  2007-07-09 00:00:00             NaN   
7 years                    A006100  2007-07-09 00:00:00             NaN   
6 years                    A006100  2007-07-09 00:00:00             NaN   
10 years                   A047759  2004-04-02 00:00:00         Partner   
16 years                   A134067  1997-10-16 00:00:00             NaN   

                     outcome_type sex_upon_outcome  age_upon_outcome_(days)  \
age_upon_outcome                                                              
10 years          Return to Owner    Neutered Male                     3650   
7 years           Return to Owner    Neutered Male                     2555   
6 years           Return to Owner    Neutered Male                     2190   
10 years                 Transfer    Neutered Male                     3650   
16 years          Return to Owner    Neutered Male                     5840   

                  age_upon_outcome_(years) age_upon_outcome_age_group  \
age_upon_outcome                                                        
10 years                              10.0                (7.5, 10.0]   
7 years                                7.0                 (5.0, 7.5]   
6 years                                6.0                 (5.0, 7.5]   
10 years                              10.0                (7.5, 10.0]   
16 years                              16.0               (15.0, 17.5]   

                     outcome_datetime  outcome_month  ...  \
age_upon_outcome                                      ...   
10 years          2017-12-07 14:07:00             12  ...   
7 years           2014-12-20 16:35:00             12  ...   
6 years           2014-03-08 17:10:00              3  ...   
10 years          2014-04-07 15:12:00              4  ...   
16 years          2013-11-16 11:54:00             11  ...   

                  age_upon_intake_age_group      intake_datetime intake_month  \
age_upon_outcome                                                                
10 years                        (7.5, 10.0]  2017-12-07 00:00:00           12   
7 years                          (5.0, 7.5]  2014-12-19 10:21:00           12   
6 years                          (5.0, 7.5]  2014-03-07 14:26:00            3   
10 years                        (7.5, 10.0]  2014-04-02 15:55:00            4   
16 years                       (15.0, 17.5]  2013-11-16 09:02:00           11   

                  intake_year  intake_monthyear  intake_weekday  intake_hour  \
age_upon_outcome                                                               
10 years                 2017           2017-12        Thursday           14   
7 years                  2014           2014-12          Friday           10   
6 years                  2014           2014-03          Friday           14   
10 years                 2014           2014-04       Wednesday           15   
16 years                 2013           2013-11        Saturday            9   

                 intake_number            time_in_shelter time_in_shelter_days  
age_upon_outcome                                                                
10 years                   1.0  0 days 14:07:00.000000000             0.588194  
7 years                    2.0  1 days 06:14:00.000000000             1.259722  
6 years                    3.0  1 days 02:44:00.000000000             1.113889  
10 years                   1.0  4 days 23:17:00.000000000             4.970139  
16 years                   1.0  0 days 02:52:00.000000000             0.119444  

[5 rows x 40 columns]

In [210]:
intakesOutcomes.shape

(79672, 40)

In [223]:
outcomes = pd.read_csv(outcomesCsv, index_col=0)
outcomes.head()

animal_id animal_type                    breed         color  \
age_upon_outcome                                                                
2 weeks            A684346         Cat   Domestic Shorthair Mix  Orange Tabby   
1 year             A666430         Dog               Beagle Mix   White/Brown   
1 year             A675708         Dog                 Pit Bull    Blue/White   
9 years            A680386         Dog  Miniature Schnauzer Mix         White   
5 months           A683115       Other                  Bat Mix         Brown   

                        date_of_birth             datetime  \
age_upon_outcome                                             
2 weeks           2014-07-07T00:00:00  2014-07-22T16:04:00   
1 year            2012-11-06T00:00:00  2013-11-07T11:47:00   
1 year            2013-03-31T00:00:00  2014-06-03T14:20:00   
9 years           2005-06-02T00:00:00  2014-06-15T15:50:00   
5 months          2014-01-07T00:00:00  2014-07-07T14:04:00   

                            monthyear     name outcome_subtype outcome_type  \
age_upon_outcome                                                              
2 weeks           2014-07-22T16:04:00      NaN         Partner     Transfer   
1 year            2013-11-07T11:47:00     Lucy         Partner     Transfer   
1 year            2014-06-03T14:20:00  *Johnny             NaN     Adoption   
9 years           2014-06-15T15:50:00   Monday         Partner     Transfer   
5 months          2014-07-07T14:04:00      NaN     Rabies Risk   Euthanasia   

                 sex_upon_outcome  
age_upon_outcome                   
2 weeks               Intact Male  
1 year              Spayed Female  
1 year              Neutered Male  
9 years             Neutered Male  
5 months                  Unknown

In [212]:
outcomes.describe(), outcomes.dtypes, outcomes.shape

(       animal_id animal_type                   breed        color  \
 count      80681       80681                   80681        80681   
 unique     72877           5                    2176          532   
 top      A721033         Dog  Domestic Shorthair Mix  Black/White   
 freq          13       45856                   23821         8396   
 
               date_of_birth             datetime            monthyear   name  \
 count                 80681                80681                80681  56116   
 unique                 5956                66474                66474  14824   
 top     2015-09-01T00:00:00  2016-04-18T00:00:00  2016-04-18T00:00:00  Bella   
 freq                    112                   39                   39    362   
 
        outcome_subtype outcome_type sex_upon_outcome  
 count            36893        80667            80679  
 unique              19            9                5  
 top            Partner     Adoption    Neutered Male  
 freq            

# Question 1

* is there an area where more pets are found?
    - Find the top 5 places where animals are found so the shelter can coordinate with local volunteers and animal control to monitor these areas.*


In [156]:
intakes['found_location'].value_counts().head(50)


found_location
Austin (TX)                                     14443
Outside Jurisdiction                              948
Travis (TX)                                       921
7201 Levander Loop in Austin (TX)                 517
Del Valle (TX)                                    411
Pflugerville (TX)                                 372
Manor (TX)                                        280
4434 Frontier Trl in Austin (TX)                  163
124 W Anderson Ln in Austin (TX)                  153
Leander (TX)                                      113
12034 Research Blvd in Austin (TX)                 98
1156 W Cesar Chavez in Austin (TX)                 98
12034 Research in Austin (TX)                      96
1834 Ferguson in Austin (TX)                       90
1156 W Cesar Chavez St in Austin (TX)              74
4106 N Lamar in Austin (TX)                        71
14811 Chicadee in Pflugerville (TX)                67
4106 N Lamar Blvd in Austin (TX)                   66
Lago Vista (T

well, I can't concentrate on a non-concentrated location so I want to stick to exclusively addesses. This means I'll want to filter out anything that doesn't start with a list of numbers.

In [169]:
# a regular expression that says return all found_locations that start with numbers
pattern = r'^\d+.*'
preciseLocations = intakes['found_location'].str.contains(pattern, regex=True)
intakes[preciseLocations]['found_location'].value_counts()



found_location
7201 Levander Loop in Austin (TX)       517
4434 Frontier Trl in Austin (TX)        163
124 W Anderson Ln in Austin (TX)        153
1156 W Cesar Chavez in Austin (TX)       98
12034 Research Blvd in Austin (TX)       98
                                       ... 
5401 Bahan Dr in Del Valle (TX)           1
4401 Leslie Ave in Austin (TX)            1
19421 Tayshas St in Manor (TX)            1
1100 West Parmer Lane in Austin (TX)      1
4605 Lyra Cir in Austin (TX)              1
Name: count, Length: 25597, dtype: int64

# Final Answer - Top 5 addresses to focus on
* 7201 Levander Loop in Austin (TX)       517
* 4434 Frontier Trl in Austin (TX)        163
* 124 W Anderson Ln in Austin (TX)        153
* 1156 W Cesar Chavez in Austin (TX)       98
* 12034 Research Blvd in Austin (TX)       98 

# Question 2
* What is the average number of pets found in a month in the year 2015? Are there months where there is a higher number of animals found?
    - Knowing the number of pets the shelter might see in a month can help them gather enough resources and donations to care for the animals they receive.

In [291]:
# In order to do any filtering I first have to convert intake datetime field to the dtype: datetime
intakes['datetime'] = pd.to_datetime(intakes['datetime'])
intakesOutcomes['intake_datetime'] = pd.to_datetime(intakesOutcomes['intake_datetime'])
# and for later
outcomes['datetime'] = pd.to_datetime(outcomes['datetime'])



In [208]:
#I need to find all the intakes in 2015
filteredData = intakes[intakes['datetime'].dt.year == 2015]
#I want to confirm the intakes.datetime and intakesOutcomes.intakedatetime are the same, so i'll merge the filtered dataview with intakeOutcomes and do a few spot checks
#better yet, I'll merge them as the same column and that will ensure we only inspect matching data
intakesIntakesOutcomes = filteredData.merge(intakesOutcomes, left_on=['animal_id','datetime'], right_on=['animal_id_outcome','intake_datetime'])
monthlyIntakeTotals2015 = intakesIntakesOutcomes['intake_month'].value_counts()
monthlyIntakeTotals2015.describe()



count      12.000000
mean     1553.750000
std       343.050785
min      1123.000000
25%      1308.000000
50%      1551.000000
75%      1720.500000
max      2176.000000
Name: count, dtype: float64

In [207]:
monthlyIntakeTotals2015.sort_index()

intake_month
1     1203
2     1123
3     1343
4     1526
5     2087
6     2176
7     1629
8     1716
9     1576
10    1734
11    1405
12    1127
Name: count, dtype: int64

In [204]:
(monthlyIntakeTotals2015 > monthlyIntakeTotals2015.mean())

intake_month
1     False
2     False
3     False
4     False
5      True
6      True
7      True
8      True
9      True
10     True
11    False
12    False
Name: count, dtype: bool

# Final Answer - Above average intake months in the year 2015
* May through October are above average
* May and June are more than 1 std above average.  I would focus my resources there.

# Question 3 
* What is the ratio of incoming pets vs. adopted pets?
    - This key metric helps the shelter know how they are doing.

In [284]:
intakesOutcomes['date_of_birth'] = pd.to_datetime(intakesOutcomes['date_of_birth'])
outcomes['date_of_birth'] = pd.to_datetime(outcomes['date_of_birth'])
#First I'd like to merge intakes and outcomes to makes sure i'm looking at the outcomes for the pets we've intaked
OutcomesAndIntakeOutcomes = intakesOutcomes.merge(outcomes, left_on=['animal_id_outcome','date_of_birth', 'animal_type','breed','color','outcome_subtype','outcome_type','sex_upon_outcome'], right_on=['animal_id','date_of_birth', 'animal_type','breed','color','outcome_subtype','outcome_type','sex_upon_outcome'])

OutcomesAndIntakeOutcomes.head()


animal_id_outcome date_of_birth outcome_subtype     outcome_type  \
0           A006100    2007-07-09             NaN  Return to Owner   
1           A006100    2007-07-09             NaN  Return to Owner   
2           A006100    2007-07-09             NaN  Return to Owner   
3           A006100    2007-07-09             NaN  Return to Owner   
4           A006100    2007-07-09             NaN  Return to Owner   

  sex_upon_outcome  age_upon_outcome_(days)  age_upon_outcome_(years)  \
0    Neutered Male                     3650                      10.0   
1    Neutered Male                     3650                      10.0   
2    Neutered Male                     3650                      10.0   
3    Neutered Male                     2555                       7.0   
4    Neutered Male                     2555                       7.0   

  age_upon_outcome_age_group     outcome_datetime  outcome_month  ...  \
0                (7.5, 10.0]  2017-12-07 14:07:00             12  ...   
1                (7.5, 10.0]  2017-12-07 14:07:00             12  ...   
2                (7.5, 10.0]  2017-12-07 14:07:00             12  ...   
3                 (5.0, 7.5]  2014-12-20 16:35:00             12  ...   
4                 (5.0, 7.5]  2014-12-20 16:35:00             12  ...   

   intake_monthyear intake_weekday intake_hour  intake_number  \
0           2017-12       Thursday          14            1.0   
1           2017-12       Thursday          14            1.0   
2           2017-12       Thursday          14            1.0   
3           2014-12         Friday          10            2.0   
4           2014-12         Friday          10            2.0   

             time_in_shelter  time_in_shelter_days  animal_id  \
0  0 days 14:07:00.000000000              0.588194    A006100   
1  0 days 14:07:00.000000000              0.588194    A006100   
2  0 days 14:07:00.000000000              0.588194    A006100   
3  1 days 06:14:00.000000000              1.259722    A006100   
4  1 days 06:14:00.000000000              1.259722    A006100   

              datetime            monthyear   name  
0  2014-03-08T17:10:00  2014-03-08T17:10:00  Scamp  
1  2014-12-20T16:35:00  2014-12-20T16:35:00  Scamp  
2  2017-12-07T00:00:00  2017-12-07T00:00:00  Scamp  
3  2014-03-08T17:10:00  2014-03-08T17:10:00  Scamp  
4  2014-12-20T16:35:00  2014-12-20T16:35:00  Scamp  

[5 rows x 44 columns]

In [285]:
#lets see all the possible outcome_types
outcomeTypes = OutcomesAndIntakeOutcomes['outcome_type'].value_counts()
outcomeTypes
adoptedToTotalRatio = outcomeTypes.loc['Adoption']/outcomeTypes.sum()
adoptedToTotalRatio

0.43330650229901535

# Final Answer - Percentage of adopted pets
43.3% of animals that are intaken animals are adopted

# Question 4
* What is the distribution of the types of animals in the shelter?
    - Find the count of each type of animal in the shelter.

In [278]:
intakes['animal_type'].value_counts()

animal_type
Dog          45743
Cat          29659
Other         4434
Bird           342
Livestock        9
Name: count, dtype: int64

# Final Answer - animal type breakdown
|animal type | count |
| ------ | ------ | 
|Dog|          45743|
|Cat|          29659|
|Bird|           342|
|Livestock|        9|


# Question 5 
* What are the adoption rates for specific breeds?
    - Find the top 5 dog breeds in the shelter (based on count) and then find the adoption percentage of each breed.

In [309]:
outcomes.dtypes

outcome_type
Adoption           34232
Transfer           24050
Return to Owner    14851
Euthanasia          6289
Died                 699
Disposal             304
Rto-Adopt            179
Missing               47
Relocate              16
Name: count, dtype: int64

In [369]:
#outcomes has both fields so lets just use it
dogBreedTotals = outcomes[outcomes['animal_type'] == 'Dog']['breed'].value_counts()
dogBreedTotals

breed
Pit Bull Mix                                       6361
Chihuahua Shorthair Mix                            4874
Labrador Retriever Mix                             4849
German Shepherd Mix                                1967
Australian Cattle Dog Mix                          1112
                                                   ... 
Australian Cattle Dog/Alaskan Husky                   1
Papillon/Pomeranian                                   1
Papillon/Yorkshire Terrier                            1
Miniature Schnauzer/Soft Coated Wheaten Terrier       1
Labrador Retriever/Pekingese                          1
Name: count, Length: 1932, dtype: int64

In [359]:
dogBreedAdoptions = outcomes[(outcomes.animal_type == 'Dog') & (outcomes.outcome_type == 'Adoption')]['breed'].value_counts()
dogBreedAdoptions

breed
Labrador Retriever Mix                     2404
Pit Bull Mix                               2382
Chihuahua Shorthair Mix                    2293
German Shepherd Mix                         937
Australian Cattle Dog Mix                   619
                                           ... 
Great Pyrenees/English Setter                 1
American Foxhound/Labrador Retriever          1
Dachshund Longhair/Pembroke Welsh Corgi       1
Cardigan Welsh Corgi/Cairn Terrier            1
Labrador Retriever/Pekingese                  1
Name: count, Length: 1390, dtype: int64

In [362]:
dogBreedAdoptionPercentage = dogBreedAdoptions/dogBreedTotals
dogBreedAdoptionPercentage

breed
Affenpinscher Mix                                0.625000
Afghan Hound Mix                                      NaN
Afghan Hound/Labrador Retriever                  1.000000
Airedale Terrier                                      NaN
Airedale Terrier Mix                             0.333333
                                                   ...   
Yorkshire Terrier/Shih Tzu                       0.666667
Yorkshire Terrier/Soft Coated Wheaten Terrier    1.000000
Yorkshire Terrier/Standard Poodle                     NaN
Yorkshire Terrier/Toy Poodle                          NaN
Yorkshire Terrier/Yorkshire Terrier              1.000000
Name: count, Length: 1932, dtype: float64

In [364]:
top5BreedsAndAdoptionRates = {'Pit Bull Mix':dogBreedAdoptionPercentage.loc['Pit Bull Mix'], 'Chihuahua Shorthair Mix': dogBreedAdoptionPercentage.loc['Chihuahua Shorthair Mix'], 'Labrador Retriever Mix':dogBreedAdoptionPercentage.loc['Labrador Retriever Mix'], 'German Shepherd Mix': dogBreedAdoptionPercentage.loc['German Shepherd Mix'], 'Australian Cattle Dog Mix': dogBreedAdoptionPercentage.loc['Australian Cattle Dog Mix']}
top5BreedsAndAdoptionRates


{'Pit Bull Mix': 0.3744694230466908,
 'Chihuahua Shorthair Mix': 0.47045547804677884,
 'Labrador Retriever Mix': 0.4957723241905548,
 'German Shepherd Mix': 0.47635993899339096,
 'Australian Cattle Dog Mix': 0.5566546762589928}

# Final Answer - Top 5 Dog Breeds and adoption ratio
* The top 5 breeds was determined by getting the value_counts of outcomes['breed']
* The adoption numbers were determined by broadcasting the boolean condition "outcome_type" = "Adoption" 
* The ratio was determined by taking the ratio of the two


| Breed | Adoption Ratio |
| ------ | ------ |
| Pit Bull Mix | 0.3744694230466908 |
| Chihuahua Shorthair Mix | 0.47045547804677884 |
| Labrador Retriever Mix | 0.4957723241905548 |
| German Shepherd Mix | 0.47635993899339096 |
| Australian Cattle Dog Mix | 0.5566546762589928 |


# Question 6 
* What are the adoption rates for different colorings?
    - Find the top 5 colorings in the shelter (based on count) and then find the adoption percentage of each color.

In [370]:
#this is basically the same as the last question but with color instead of Breed
dogColorTotals = outcomes[outcomes['animal_type'] == 'Dog']['color'].value_counts()
dogColorTotals

color
Black/White                    5276
Black                          2547
Brown/White                    2503
Tan/White                      2477
White                          2387
                               ... 
Brown Brindle/Brown Brindle       1
Tan/Apricot                       1
White/Brown Tiger                 1
Black Smoke/Black                 1
Red Merle/Red Merle               1
Name: count, Length: 339, dtype: int64

In [371]:
dogColorAdoptions = outcomes[(outcomes.animal_type == 'Dog') & (outcomes.outcome_type == 'Adoption')]['color'].value_counts()
dogColorAdoptions

color
Black/White               2610
Tan/White                 1178
Brown/White               1123
Black                     1089
Tricolor                   969
                          ... 
Gold/Cream                   1
Tricolor/Brown Brindle       1
Cream/Silver                 1
Gold/Black                   1
Black/Blue                   1
Name: count, Length: 260, dtype: int64

In [373]:
dogColorAdoptionPercentage = dogColorAdoptions/dogColorTotals
dogColorAdoptionPercentage

color
Agouti              0.500000
Apricot             0.527273
Apricot/Brown            NaN
Apricot/Tricolor         NaN
Apricot/White       0.300000
                      ...   
Yellow/Cream        0.500000
Yellow/Gray              NaN
Yellow/Tan          0.500000
Yellow/White        0.432432
Yellow/Yellow            NaN
Name: count, Length: 339, dtype: float64

In [376]:
top5ColorAndAdoptionRates = {'Black/White':dogColorAdoptionPercentage.loc['Black/White'], 'Tan/White': dogColorAdoptionPercentage.loc['Tan/White'], 'Brown/White':dogColorAdoptionPercentage.loc['Brown/White'], 'Black': dogColorAdoptionPercentage.loc['Black'], 'Tricolor': dogColorAdoptionPercentage.loc['Tricolor']}
top5ColorAndAdoptionRates

{'Black/White': 0.49469294920394236,
 'Tan/White': 0.47557529269277354,
 'Brown/White': 0.44866160607271277,
 'Black': 0.4275618374558304,
 'Tricolor': 0.4881612090680101}

# Final Answer - Top 5 color and adoption rate
the logic mirrors the last problem exactly
| color | Adoption Rate |
| ------ | ------ |
|Black/White| 0.49469294920394236|
 |Tan/White| 0.47557529269277354|
 |Brown/White| 0.44866160607271277|
 |Black| 0.4275618374558304|
 |Tricolor| 0.4881612090680101|

# Question 7
* About how many animals are spayed/neutered each month?
    - This will help the shelter allocate resources and staff. Assume that all intact males and females will be spayed/neutered.

In [419]:
intakes.dtypes, intakesOutcomes.dtypes, outcomes.dtypes

(animal_id                   object
 animal_type                 object
 breed                       object
 color                       object
 datetime            datetime64[ns]
 datetime2                   object
 found_location              object
 intake_condition            object
 intake_type                 object
 name                        object
 sex_upon_intake             object
 dtype: object,
 animal_id_outcome                     object
 date_of_birth                 datetime64[ns]
 outcome_subtype                       object
 outcome_type                          object
 sex_upon_outcome                      object
 age_upon_outcome_(days)                int64
 age_upon_outcome_(years)             float64
 age_upon_outcome_age_group            object
 outcome_datetime                      object
 outcome_month                          int64
 outcome_year                           int64
 outcome_monthyear                     object
 outcome_weekday                    

In [420]:
#what is the breakdown of sex_upon_intake?
intakes['sex_upon_intake'].value_counts()


sex_upon_intake
Intact Male      25488
Intact Female    23860
Neutered Male    12794
Spayed Female    11231
Unknown           6813
Name: count, dtype: int64

In [421]:
intactMaleOrFemaleCondition = (intakes['sex_upon_intake'] =='Intact Male') | (intakes['sex_upon_intake'] == 'Intact Female')
intactMaleOrFemale =intakes[intactMaleOrFemaleCondition]
monthlyIntakeData = intactMaleOrFemale.merge(intakesOutcomes, left_on=['animal_id','datetime','sex_upon_intake'], right_on=['animal_id_outcome','intake_datetime','sex_upon_intake'])
monthlyIntakeData['intake_month'].value_counts().sort_index().mean()


4071.0833333333335

# Final Answer - How many spayed/neutered?

It follows from the assumption that all intake male and females are neutered that I should be using 'intake_month' counts to detemine the mean value of males and females neutered.  First, I grabbed only the intake records where "sex_upon_intake" was either Male or Female intact and merged with intakeOutcomes to get all the corresponding data for months.  Then I averaged the month counts.  

4071.0833333333335